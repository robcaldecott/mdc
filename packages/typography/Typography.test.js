import React from "react";
import {
  Typography,
  Display4,
  Display3,
  Display2,
  Display1,
  Headline,
  Title,
  Subheading2,
  Subheading1,
  Body2,
  Body1,
  Caption,
  Button
} from ".";
import renderer from "react-test-renderer";

it("renders", () => {
  expect(
    renderer.create(
      <Typography element="h1" type="display4" adjustMargin>
        Display4
      </Typography>
    )
  ).toMatchSnapshot();
});

it("renders display4", () => {
  expect(renderer.create(<Display4>Display4</Display4>)).toMatchSnapshot();
});

it("renders display3", () => {
  expect(renderer.create(<Display3>Display3</Display3>)).toMatchSnapshot();
});

it("renders display2", () => {
  expect(renderer.create(<Display2>Display2</Display2>)).toMatchSnapshot();
});

it("renders display1", () => {
  expect(renderer.create(<Display1>Display4</Display1>)).toMatchSnapshot();
});

it("renders headline", () => {
  expect(renderer.create(<Headline>Headline</Headline>)).toMatchSnapshot();
});

it("renders title", () => {
  expect(renderer.create(<Title>Title</Title>)).toMatchSnapshot();
});

it("renders subheading2", () => {
  expect(
    renderer.create(<Subheading2>Subheading2</Subheading2>)
  ).toMatchSnapshot();
});

it("renders subheading1", () => {
  expect(
    renderer.create(<Subheading1>Subheading1</Subheading1>)
  ).toMatchSnapshot();
});

it("renders body2", () => {
  expect(renderer.create(<Body2>Body2</Body2>)).toMatchSnapshot();
});

it("renders body1", () => {
  expect(renderer.create(<Body1>Body1</Body1>)).toMatchSnapshot();
});

it("renders caption", () => {
  expect(renderer.create(<Caption>Caption</Caption>)).toMatchSnapshot();
});

it("renders button", () => {
  expect(renderer.create(<Button>Button</Button>)).toMatchSnapshot();
});
