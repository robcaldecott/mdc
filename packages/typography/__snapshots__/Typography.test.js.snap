// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<h1
  className="mdc-typography--display4 mdc-typography--adjust-margin"
>
  Display4
</h1>
`;

exports[`renders body1 1`] = `
<p
  className="mdc-typography--body1"
>
  Body1
</p>
`;

exports[`renders body2 1`] = `
<aside
  className="mdc-typography--body2"
>
  Body2
</aside>
`;

exports[`renders button 1`] = `
<span
  className="mdc-typography--button"
>
  Button
</span>
`;

exports[`renders caption 1`] = `
<span
  className="mdc-typography--caption"
>
  Caption
</span>
`;

exports[`renders display1 1`] = `
<h1
  className="mdc-typography--display1"
>
  Display4
</h1>
`;

exports[`renders display2 1`] = `
<h1
  className="mdc-typography--display2"
>
  Display2
</h1>
`;

exports[`renders display3 1`] = `
<h1
  className="mdc-typography--display3"
>
  Display3
</h1>
`;

exports[`renders display4 1`] = `
<h1
  className="mdc-typography--display4"
>
  Display4
</h1>
`;

exports[`renders headline 1`] = `
<h1
  className="mdc-typography--headline"
>
  Headline
</h1>
`;

exports[`renders subheading1 1`] = `
<h4
  className="mdc-typography--subheading1"
>
  Subheading1
</h4>
`;

exports[`renders subheading2 1`] = `
<h3
  className="mdc-typography--subheading2"
>
  Subheading2
</h3>
`;

exports[`renders title 1`] = `
<h1
  className="mdc-typography--title"
>
  Title
</h1>
`;
